lambda_src/lambda_function.py

```diff
@@ -143,9 +152,65 @@ def sync_flow(event: Any, context: Any = None) -> Dict[Text, Any]:
                 row_result,
             ]
         )
+    return res_data
+
+
+def sync_flow(event: Any, context: Any = None) -> ResponseType:
+    """
+    Handles the synchronous part of the generic lambda flows.
+
+    Args:
+        event (Any): This the event object as received by the lambda_handler()
+        context (Any): Has the function context. Defaults to None.
+
+    Returns:
+        ResponseType: Represents the response status and data.
+    """
+    LOG.debug('Destination header not found in a POST and hence using sync_flow().')
+    headers = event['headers']
+    req_body = loads(event['body'])
+    start_time = timer()
+
+    destination_driver = None
+    batch_id = headers[BATCH_ID_HEADER]
+    write_uri = headers.get('write-uri')
+    destination_driver = (
+        import_module(f'geff.drivers.destination_{urlparse(write_uri).scheme}')
+        if write_uri
+        else None
+    )
+
+    LOG.debug(f'sync_flow() received destination: {write_uri}.')
+
+    if not destination_driver:
+        if not is_batch_initialized(batch_id):
+            initialize_batch(batch_id)
+        else:
+            while is_batch_processing(batch_id):
+                if (timer() - start_time) > 30:
+                    error_dumps = dumps(
+                        {
+                            'data': [
+                                [
+                                    rn,
+                                    {'error': 'API gateway timed out.'},
+                                ]
+                                for rn, *args in req_body['data']
+                            ]
+                        }
+                    )
+                    gateway_timeout_response = {
+                        'statusCode': 200,
+                        'body': error_dumps,
+                    }
+                    finish_batch_processing(batch_id, gateway_timeout_response)
```

Comment:
I think you can replace this with a `return`, as long as the API Gateway times out in 30s, the lambda invocation will be abandoned / retried and it doesn't matter what you respond.